test/unit/transform/eia861.py

```diff
@@ -0,0 +1,107 @@
+"""Unit tests for the pudl.transform.eia861 module."""
+
+from io import StringIO
+
+import pandas as pd
+import pytest
+
+from pudl.metadata.fields import apply_pudl_dtypes
+from pudl.transform import eia861
+
+# Test that rows with the same primary key are combined correctly
```

Comment:
Nice work with setting up the parametrized tests and adding comments to explain each df!